idseq_pipeline/commands/non_host_alignment_functions.py

```diff
@@ -353,80 +380,104 @@ def get_map(m8_file):
+        # * alignment_length = length of the alignments
+        # * e_value = the expect value
+        #
+        # See:
+        # * http://www.metagenomics.wiki/tools/blast/blastn-output-format-6
+        # * http://www.metagenomics.wiki/tools/blast/evalue
+
         m8_line_columns = m8_line.split("\t")
-        assert m8_line_columns[0] == hit_line_columns[
-            0], "read_ids in %s and %s do not match: %s vs. %s" % (
-                os.path.basename(m8_file), os.path.basename(hit_level_file),
-                m8_line_columns[0], hit_line_columns[0])
+        msg = "read_ids in %s and %s do not match: %s vs. %s" % (
+            os.path.basename(m8_file), os.path.basename(hit_level_file),
+            m8_line_columns[0], hit_line_columns[0])
+        assert m8_line_columns[0] == hit_line_columns[0], msg
         percent_identity = float(m8_line_columns[2])
         alignment_length = float(m8_line_columns[3])
         e_value = float(m8_line_columns[10])
+
         # These have been filtered out before the creation of m8_f and hit_f
         assert alignment_length > 0
         assert -0.25 < percent_identity < 100.25
         assert e_value == e_value
         if e_value_type != 'log10':
-            assert MIN_NORMAL_POSITIVE_DOUBLE <= e_value  # Subtle, but this excludes positive subnorms.
+            # Subtle, but this excludes positive subnormal numbers.
+            assert MIN_NORMAL_POSITIVE_DOUBLE <= e_value
             e_value = math.log10(e_value)
 
-        # Retrieve the taxon lineage and mark meaningless calls with fake taxids.
+        # Retrieve the taxon lineage and mark meaningless calls with fake
+        # taxids.
         hit_taxids_all_levels = lineage_map.get(hit_taxid, NULL_LINEAGE)
         cleaned_hit_taxids_all_levels = validate_taxid_lineage(
             hit_taxids_all_levels, hit_taxid, hit_level)
         assert NUM_RANKS == len(cleaned_hit_taxids_all_levels)
 
-        if not any_hits_to_remove(cleaned_hit_taxids_all_levels):
-            # Aggregate each level
-            agg_key = tuple(cleaned_hit_taxids_all_levels)
-            while agg_key:
-                agg_bucket = aggregation.get(agg_key)
-                if not agg_bucket:
-                    agg_bucket = {
-                        'count': 0,
-                        'sum_percent_identity': 0.0,
-                        'sum_alignment_length': 0.0,
-                        'sum_e_value': 0.0
-                    }
-                    aggregation[agg_key] = agg_bucket
-                agg_bucket['count'] += 1
-                agg_bucket['sum_percent_identity'] += percent_identity
-                agg_bucket['sum_alignment_length'] += alignment_length
-                agg_bucket['sum_e_value'] += e_value
-                # Chomp off the lowest rank as we aggregate up the tree
-                agg_key = agg_key[1:]
+        if any_hits_to_remove(cleaned_hit_taxids_all_levels):
+            hit_line = hit_f.readline()
```

Comment:
not sure about this change. now there's a bit of code dupe...

Reply:
changed back